packages/babel-parser/src/plugins/typescript.js

```diff
@@ -747,6 +753,22 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         this.next();
         return true;
       }
+
+      if (this.match(tt.braceL)) {
+        const braceStack = [tt.braceL];
```

Comment:
This can be an integer instead of an array, since the check only uses it's length.

Reply:
changed